crates/wasm-smith/src/lib.rs

```diff
@@ -1665,6 +1739,31 @@ impl Module {
         })
     }
 
+    fn arbitrary_tag_type(&self, u: &mut Unstructured) -> Result<TagType> {
+        let candidate_func_types = self.tag_func_types().collect::<Vec<_>>();
+        let max = candidate_func_types.len() - 1;
+        let ty = candidate_func_types[u.int_in_range(0..=max)?];
+        Ok(TagType {
+            func_type_idx: ty,
+            func_type: self.func_type(ty).clone(),
+        })
+    }
+
+    fn arbitrary_tags(&mut self, u: &mut Unstructured) -> Result<()> {
+        if !self.config.exceptions_enabled() || self.has_tag_func_types() {
```

Comment:
I'm not familiar with the exceptions proposal yet, but is this supposed to be

```suggestion
        if !self.config.exceptions_enabled() || !self.has_tag_func_types() {
```

?

Reply:
Yes, indeed. Good catch.